controllers/cluster_controller.go

```diff
@@ -640,31 +642,60 @@ func checkCoordinatorValidity(cluster *fdbtypes.FoundationDBCluster, status *fdb
 			continue
 		}
 
-		var address string
+		var ipAddress fdbtypes.ProcessAddress
 		for _, addr := range addresses {
 			if addr.Flags["tls"] == cluster.Spec.MainContainer.EnableTLS {
-				address = addr.String()
+				ipAddress = addr
 				break
 			}
 		}
 
-		_, isCoordinator := coordinatorStatus[address]
-		if isCoordinator && !process.Excluded && !pendingRemoval {
-			coordinatorStatus[address] = true
+		coordinatorAddress := ""
+		_, isCoordinatorWithIP := coordinatorStatus[ipAddress.String()]
+		if isCoordinatorWithIP {
+			coordinatorAddress = ipAddress.String()
 		}
 
-		if isCoordinator {
+		dnsName := process.Locality["dns_name"]
```

Comment:
```suggestion
		dnsName := process.Locality["dns_name"]
```

Can we use the constant `FDBLocalityDNSNameKey` here?